ykrt/src/deopt.rs

```diff
@@ -0,0 +1,59 @@
+//! Run-time deoptimisation support: when a guard fails, this module restores the state necessary
+//! to resume interpreter execution.
+
+use std::{arch::asm, ffi::c_void};
+
+/// The `__llvm__deoptimize()` function required by `llvm.experimental.deoptimize` intrinsic, that
+/// we use for exiting to the stop-gap interpreter on guard failure.
```

Comment:
mention of stop-gap interpreter here.

Reply:
There are quite a lot of "stop-gap"s in the repo, including in test file names. If we like this PR, I will do another PR which goes through and fixes all those (I worried that doing so here would make the PR hard to read).

Reply:
sounds good!